gamedata.py

```diff
@@ -1,4 +1,5 @@
 from __future__ import annotations
+from abc import abstractmethod
```

Comment:
This should be `from abc import ABC, abstractmethod`. And I would move that import between `import io` and `from aiohttp.web ...` with newlines above and below (this is just a stylistic choice).

Additionally, `FileParser` should inherit `ABC`.